File: final/1st.cpp
#include <iostream>
#include <ranges>
#include <set>
#include <algorithm>


int main(int argc, char const *argv[])
{
    int n;

    std::cin >> n;
    unsigned long long m = 0;
    for (int i = 0, j = 0, total = 0; total < n+1;++total)
    {
        unsigned long long sq = (unsigned long long)i * i;
        unsigned long long qr = (unsigned long long)j * j * j;
        m = sq < qr ? sq : qr;
        if (sq == qr)
        {
            ++i;
            ++j;
        }
        else if (sq > qr)
            ++j;
        else
            ++i;
        
    }
    std::cout << m << std::endl;
    return 0;
}

File: final/1st.py
n = int(input())
m = 0
i = 0
j = 0
total = 0

while total < n + 1:
    sq = i * i
    qr = j * j * j
    m = min(sq, qr)
    
    if sq == qr:
        i += 1
        j += 1
    elif sq > qr:
        j += 1
    else:
        i += 1
    
    total += 1

print(m)
